src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -662,5 +662,24 @@ public static bool SequenceEqual<T>(this IEnumerable<T>? first, IEnumerable<T>?
                 return result;
             }
         }
+
+        public static TResult Aggregate<TSource, TSeed, TResult>(this IEnumerable<TSource> source, TSeed seed, Func<TSeed, TSource, TResult> seedSelector, Func<TResult, TSource, TResult> resultSelector)
+        {
+            using (var e = source.GetEnumerator())
+            {
+                if (!e.MoveNext())
+                {
+                    throw new InvalidOperationException();
+                }
```

Comment:
not sure i buy that this is invalid.  i woudl expect aggregation to work on an empty source.  it would just translate teh seed to the result.

Reply:
There are two variation of Aggregate in BLC, one that doesn't take a seed and requires a non-empty source, and one that takes a seed and does work on empty source. I created AggregateOrDefault a while back (right above this method) to relax the first variation to work on an empty list. However, when the seed is of a different type than the result, we do need a non-empty list. This helper is covering that case.